worker/jwtparser/worker.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jwtparser
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/juju/worker/v3"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/worker/jwtparser/jwt"
+)
+
+type jwtParserWorker struct {
+	tomb      tomb.Tomb
+	jwtParser *jwt.JWTParser
+}
+
+func newWorker(statePool *state.StatePool) (worker.Worker, error) {
+	systemState, err := statePool.SystemState()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	controllerConfig, err := systemState.ControllerConfig()
+	if err != nil {
+		return nil, errors.Annotate(err, "cannot fetch the controller config")
+	}
+	jwtRefreshURL := controllerConfig.LoginTokenRefreshURL()
```

Comment:
Can the refresh url change? if that's the case, should it be part of the main loop?

Reply:
Currently it can't, the value can only be set at bootstrap time. In a follow-up, I will make it changeable at runtime and make this part of the main loop.